src/racing/actors.c

```diff
@@ -2377,16 +1284,17 @@ void func_8029E7DC(struct Actor *actor) {
  * @param Actor to destroy
  */
 void destroy_actor(struct Actor *actor) {
-    func_8029E7DC(actor);
+    play_sound_before_destroy(actor);
     actor->flags = 0;
     actor->type = 0;
     gNumActors--;
 }
 
-s16 func_8029E890(Vec3f pos, Vec3s rot, Vec3f velocity, s16 actorType) {
+s16 try_remove_destructable_item(Vec3f pos, Vec3s rot, Vec3f velocity, s16 actorType) {
     s32 actorIndex;
     struct ShellActor *compare;
 
+    // try remove a red shell or green shell or banana or fake item box if the actor are expired
```

Comment:
// try removing a red shell, green shell, banana, or a fake item box if the actor is expired